devolutions-gateway/src/http/controllers/jet.rs

```diff
@@ -106,29 +114,38 @@ impl JetController {
         };
 
         // check the session token is signed by our provider if unrestricted mode is not set
-        if !self.config.unrestricted {
-            match validate_session_token(self.config.as_ref(), &req) {
-                Err(e) => {
-                    slog_scope::error!("Couldn't validate session token: {}", e);
-                    return (StatusCode::UNAUTHORIZED, None);
-                }
-                Ok(expected_id) if expected_id != association_id => {
+        let jet_tp_claim = match validate_session_token(self.config.as_ref(), &req) {
```

Comment:
I'm not sure just by looking at the code what would happen if the jet_tp claim is absent from the token? Devolutions Gateway needs to function properly with older versions of the code. If the jet_tp claim is omitted, it can be assumed not to record anything. Since the tokens are signed by Wayk Bastion anyway, there is no issue with trust. only versions of Wayk Bastion that can enable session recording will set this field, older versions will omit it.

Reply:
yes, the jet_tp claim has the type Option<>. So if the jet_tp claim will be absent, the value will be set to None and the recording will not start.